play-json/shared/src/main/scala/Json.scala

```diff
@@ -414,6 +419,29 @@ object Json extends JsonFacade {
   sealed trait MacroOptions
 
   /**
+   * Defines the default macro options.
+   *
+   * Since you can't have defaults for a type parameter (unless it's a contravariant type, it will default to
+   * Nothing), we supply the default via an implicit parameter.
+   */
+  trait DefaultMacroOptions[O <: Json.MacroOptions]
+
+  /**
+   * Low priority implicit used when Json.MacroOptions is not passed.
+   */
+  trait NotDefaultMacroOptions {
+    implicit def notDefault[O <: Json.MacroOptions]: DefaultMacroOptions[O] = new DefaultMacroOptions[O] {}
+  }
+
+  object DefaultMacroOptions extends NotDefaultMacroOptions {
+
+    /**
+     * This wil be the default that's passed, causing Json.MacroOptions to be the default options.
+     */
+    implicit object defaultMacroOptions extends DefaultMacroOptions[Json.MacroOptions]
+  }
+
+  /**
```

Comment:
The comments pretty much say it all, but just to clarify, consider below, the:

```scala
implicit def default[Opts <: Json.MacroOptions]: JsonConfiguration.Aux[Json.MacroOptions]`
```

on `JsonConfiguration`. If I just invoke that without supplying a type (as gets down when it's invoke implicitly), what type do I get back?  The answer is, `JsonConfiguration.Aux[Nothing]`. And what does that mean? I'm not 100% sure, but I think it means that every option is enabled, since `Nothing` is a subclass of everything. That's a pretty serious bug. Ideally, I'd like to change the signature to:

```scala
implicit def default[Opts <: Json.MacroOptions = Json.MacroOptions]: JsonConfiguration.Aux[Json.MacroOptions]`
```

ie, supply a default type parameter, but Scala doesn't support that feature. So instead we supply the default via an implicit. This solution comes courtesy of @travisbrown (thanks!):

https://stackoverflow.com/questions/25127140/how-can-parameters-settings-be-passed-to-a-scala-macro